Shopify/shopify-cli
# Add wasm to languages and project creator

### WHY are these changes introduced?

Fixes https://github.com/Shopify/script-service/issues/4122

### WHAT is this pull request doing?

Adds "wasm" as a language to script project creation.

Since "wasm" should be available to all APIs, I've opted to generate a fake/minimal config for the language when we populate all the languages for a type. As opposed to requiring "wasm" to be explicitly defined for each API in the extension_points.yml file.

### Things blocking this PR from shipping
- [x] [scripts-api-examples](https://github.com/Shopify/scripts-apis-examples/tree/add-other-language-examples/checkout/wasm) default code for "wasm" projects should be merged into main.

### How to test your changes?

Ensure you have scripts_beta_languages enabled.
Create a script, and use the `--branch=add-other-language-examples` argument to checkout the base files.
When creating, select the "wasm" language.

### Post-release steps



### Update checklist
